packages/rocketchat-message-snippet/server/requests.js

```diff
@@ -0,0 +1,47 @@
+/* global Cookies */
+WebApp.connectHandlers.use('/snippet/download', function(req, res) {
+	var match = /^\/([^\/]+)\/(.*)/.exec(req.url);
+
+	if (match[1]) {
+		let snippet = RocketChat.models.Messages.findOneById(match[1]);
```

Comment:
You should do this find after login verification, cuz it's not necessary if user is not authorized.